x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/LabelFieldProducer.java

```diff
@@ -140,8 +148,40 @@ public void reset() {
 
     static class AggregateMetricFieldProducer extends LabelLastValueFieldProducer {
 
-        AggregateMetricFieldProducer(String name, Metric metric) {
-            super(name, new LastValueLabel(metric.name()));
+        AggregateMetricFieldProducer(final MappedFieldType fieldType, final String name, final Metric metric) {
+            super(fieldType, name, new LastValueLabel(metric.name()));
+        }
+    }
+
+    static class HistogramLabelFieldProducer extends LabelLastValueFieldProducer {
```

Comment:
maybe add jdocs why this specialised subclass is needed to produce last labels for histograms?